NO MRG, TST: Re-render with CircleCI matrix

jakirkham opened the pull request:
Please do **not** merge this PR. It is for testing purposes only.

Made changes to `conda-smithy` to try and support CircleCI 2.0's workflow format. In particular, this tries to use a matrix based on their parallel workflow. More details in the PR ( https://github.com/conda-forge/conda-smithy/pull/532 ).

jakirkham committed:
NO MRG, TST: Re-render with CircleCI 2.0 support

Please do **not** merge this PR. It is for testing purposes only.

Made changes to `conda-smithy` to try and support CircleCI 2.0's format
and layout. This should make it easier to explore parallel matrix builds
in the future as well.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
NO MRG, TST: Re-render with CircleCI matrix

Please do **not** merge this PR. It is for testing purposes only.

Made changes to `conda-smithy` to try and support CircleCI 2.0's
workflow format. In particular, this tries to use a matrix based on
their parallel workflow.
